lib/ecto/query.ex

```diff
@@ -1188,8 +1188,12 @@ defmodule Ecto.Query do
     supplier_query = Supplier |> select([s], s.city)
     Customer |> select([c], c.city) |> union(supplier_query)
   """
-  defmacro union(query, other_query) do
-    Union.build(:union, query, other_query, __CALLER__)
+  def union(%Ecto.Query{unions: unions} = query, %Ecto.Query{} = other_query) do
```

Comment:
What happens if `other_query` has LIMIT / OFFSET / ORDER BY. Maybe we should check that those values have not been set and raise if they are set?